app/models/user.rb

```diff
@@ -0,0 +1,12 @@
+class User < ApplicationRecord
+	has_many :tasks, dependent: :destroy
+	validates :firstname, presence: true
+	validates :lastname, presence: true
+	validates :email, presence: true,  format: { with: URI::MailTo::EMAIL_REGEXP }, uniqueness: { case_sensitive: false }
```

Comment:
Please consider changing emails to lowercase before saving them into DB.

Reply:
Please put a unique index on the column if you are putting the uniqueness validation. It will make the operation faster otherwise all the rows will be scaned.